priv/templates/nimble_template/lib/otp_app_web/helpers/icon_helper.ex.eex

```diff
@@ -5,20 +5,17 @@ defmodule <%= web_module %>.IconHelper do
 
   use Phoenix.HTML
 
-  alias <%= web_module %>.Router.Helpers, as: Routes
+  use <%= web_module %>, :verified_routes
 
   @svg_shape_id_prefix "icon-priv--static--images--icons--"
 
-  def icon_tag(conn, icon_file_name, opts \\ []) do
+  def icon_tag(icon_file_name, opts \\ []) do
     classes = "icon " <> Keyword.get(opts, :class, "")
 
     content_tag(:svg, class: classes) do
       tag(:use,
         "xlink:href":
-          Routes.static_path(
-            conn,
-            "/images/icon-sprite.svg#" <> @svg_shape_id_prefix <> icon_file_name
-          )
+        ~p"/images/icon-sprite.svg#" <> "#{@svg_shape_id_prefix}#{icon_file_name}"
```

Comment:
Follow up with your concern `Not sure there’s any benefit using verifiedroute for static dynamic generated paths?`

Can we just use the `String` here @liamstevens111 

```suggestion
        "/images/icon-sprite.svg#" <> "#{@svg_shape_id_prefix}#{icon_file_name}"
```

Have a test on your local to see if it could render the SVG icon or not 👌

Reply:
Yeah it renders and gives warnings for `/images` if it's not in static path, that's it. Maybe look in future to redefining it as 

~p"/images/#{"icon-sprite.svg#" <> @svg_shape_id_prefix <> icon_file_name}"

But the `#` gets encoded as `%23`